### PR TITLE
examples: refactored demos for adversarial patches

### LINKED ISSUE #152: Examples: Update documentation and support files for examples to account for new deployment documentation
# Definition of Done
- [ ] Documentation landing page for examples
- [ ] Documentation updated to point to deployment documentation
- [x] Spurious files removed

READMEs answer the question of "What do I do next?"
Documentation of example will walk a user through the particulars of an example. Tutorial of the example.
Documentation page for examples will describe how to accomplish a particular task or set of tasks or describes the nature of examples/plugins/etc..

### Prerequisites Setup

1. point to deployment instructions
    - (mention cpu/gpu distinction)
2. point to dataset download instructions
3. Instructions for mounting datasets for a local/single machine installation
    - point to more general instruction.
5. Instructions for minimal python env/dependencies (point to script)
6. Usage Instructions
    - (port make file to python script)

### Table Summarizing Example
Dataset used (prereq)
GPU needed? (prereq)
What does it show? Attacks/ Defenses (info)
ML Librrary? pytorch/tensorflow (info)
architecture (info)

## Each of the examples is updated:

- [x] pytorch-detectron2-demo
- [x] pytorch-mnist-membership-inference
- [x] tensorflow-adversarial-patches
- [x] tensorflow-backdoor-poisoning
- [x] tensorflow-imagenet-pixel-threshold
- [x] tensorflow-imagenet-resnet50-fgm
- [x] tensorflow-mnist-classifier
- [x] tensorflow-mnist-feature-squeezing
- [x] tensorflow-mnist-model-inversion
- [x] tensorflow-mnist-pixel-threshold

## These examples are marked for deletion

- [x] basic-mlflow-demo
- [x] patch-defended-pixel-threshold
- [x] pytorch-robust-dpatch
- [x] tensorflow-roadsigns-yolo

### Tasks for each example
- Task plugins go into common folder under examples directory
- Script to register all task plugins
- rename to avoid collisions and indicate example
- move MLproject files to src and remove use of relative path to src
- Delete extraneous files
    - init.db
    - environment.yml
    - architecture diagram
    - utils.py
    - /data subdir
    - /jupyter subdir
- simplify notebooks to focus only on example
- remove any descriptions of dataset

### PR DESCRIPTION
**Pull request by lbarbMITRE**
Partially closes #152

### CODE REVIEW

File: examples/task-plugins/dioptra_custom/custom_patch_plugins/import_keras.py
Comment (jkglasbrenner) [inline examples/task-plugins/dioptra_custom/custom_patch_plugins/import_keras.py:1]: This is a duplicate of task-plugins/dioptra_custom/evaluation/import_keras.py, so any calls to one of the plugins in this file can be replaced with the `evaluation` version.

File: examples/task-plugins/dioptra_custom/custom_patch_plugins/tensorflow.py
Comment (jkglasbrenner) [inline examples/task-plugins/dioptra_custom/custom_patch_plugins/tensorflow.py:1]: The functions in lines 62-101 are exact duplicates of task-plugins/dioptra_custom/evaluation/tensorflow.py, and so these can be removed and the evaluation versions used instead. The function `register_init_model` is new, so this function should be kept for now.

File: examples/tensorflow-adversarial-patches/demo-fruits360-patches.ipynb
Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-fruits360-patches.ipynb:1]: Remove the section "Querying the MLFlow Tracking Service".

File: examples/tensorflow-adversarial-patches/demo-imagenet-patches.ipynb
Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-imagenet-patches.ipynb:1]: Remove the section "Querying the MLFlow Tracking Service".

Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-imagenet-patches.ipynb:1]: Update references to DATASET_DIR to point to the updated folder names.

Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-imagenet-patches.ipynb:1]: The ImageNet folder will have the name `ImageNet-Kaggle` (drop the 2017)

Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-imagenet-patches.ipynb:1]: Remove the "Dataset" section from close to top of notebook

File: examples/tensorflow-adversarial-patches/demo-mnist-patches.ipynb
Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-mnist-patches.ipynb:1]: Update DATASET_DIR references to all point to the new dataset folders

Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-mnist-patches.ipynb:1]: Delete the sentence "The MNIST, fruits360, and imagenet datasets can be used in this demo and we can choose which dataset to use."

Comment (jkglasbrenner) [inline examples/tensorflow-adversarial-patches/demo-mnist-patches.ipynb:1]: Remove the section "Querying the MLFlow Tracking Service".

**Review by jkglasbrenner — DISMISSED** [review]
In addition to the notes on the files, also be sure to address the following:

- For entrypoints that have if/else logic to create branching paths, handle this by splitting it into multiple entrypoints, one for each branch.
- Encapsulate any logic in the branching paths into a task plugin so it can be called properly.
- The branching logic described above is similar to what was handled in the backdoor poisoning examples. See the branch https://github.com/usnistgov/dioptra/blob/abyrne-update-tensorflow-backdoor-poisoning for ideas on how to resolve this.

**Comment by lbarbMITRE** [comment]
I have tried and failed to get Dioptra to run on my personal machine. A quick run to make sure the three demos work with the new task-plugin format is necessary.

**Comment by lbarbMITRE** [comment]
With feedback from Alex and Julian, the demos now run to completion (both locally and on the DGX for those requiring). Should be ready to merge.

**Review by jkglasbrenner — APPROVED** [review]

**Pull request merged**